crates/radix-engine-toolkit/src/transaction_types/interface.rs

```diff
@@ -300,6 +302,29 @@ pub fn execution_summary(
         royalty_cost: receipt.fee_summary.total_royalty_cost_in_xrd,
     };
 
+    for (node_id, states_to_update) in
```

Comment:
I think that it would be better if we avoid doing any `expect_` calls in here, mainly for two reasons:

1. We would like to minimize the number of code branches where panics can happen.
2. The `TransactionTypesReceipt` type already stores the `CommitResult`. So, there is no need to perform this `expect` again.

Additionally, can the functionality to get the newly created non-fungibles be implemented as a method on the `TransactionTypesReceipt` type itself?